cuda_core/tests/test_linker.py

```diff
@@ -46,24 +47,18 @@ def compile_ltoir_functions(init_cuda):
     return object_code_a_ltoir, object_code_b_ltoir, object_code_c_ltoir
 
 
-culink_options = [
+options = [
     LinkerOptions(),
     LinkerOptions(arch=ARCH, verbose=True),
     LinkerOptions(arch=ARCH, max_register_count=32),
     LinkerOptions(arch=ARCH, optimization_level=3),
     LinkerOptions(arch=ARCH, debug=True),
     LinkerOptions(arch=ARCH, lineinfo=True),
-    LinkerOptions(arch=ARCH, no_cache=True),  # TODO: consider adding cuda 12.4 to test matrix in which case this
-    # will fail. Tracked in issue #337
 ]
+if not culink_backend:
```

Comment:
Suggested for **line 22 above**:

```
if culink_backend:
    nvjitlink = None
else:
    from cuda.bindings import nvjitlink
```

Then here:

```
if nvjitlink is not None:
```

That way the `import` appears only once, and near the top, where it is more expected.